### PR TITLE
Upgraded all projects to VS2017 and WebRTC to M62

### PR DESCRIPTION
**Pull request by anderm**
## Checklist
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] If applicable, the code is properly documented.
- [x] The code builds without any errors.
- [x] All unit and integration tests pass.

## Description
- Upgraded all projects to VS2017 (build tools 141)
- Added a new webrtc patch for M62 build
- Removed local copy of JSON header files and switched to using the internal webrtc headers and
objects.
- Added new pre-built library in setup.cmd
- Added new libDirectXTK library for M62

## Related Issues and PRs
Related to #178 and #194 

This PR is currently blocked until the following issues are resolved:
- [x] #228 - required for CI/CD build
- [x] #194 - UWP M62 library required to fully test HoloLens client

### CODE REVIEW

**Comment by phongcao** [comment]
@anderm The DirectX Win32 client doesn't seem to work. I tried using old client and it works with our new server. Any idea?

**Comment by bengreenier** [comment]
this finally is getting a green build with the new ci! https://dpeted.visualstudio.com/TED%20Strategic%20Engagements%20(Agile)/_build/index?buildId=16383&_a=summary

**Comment by anderm** [comment]
@bengreenier @phongcao This PR is now valid and ready to be merged. Please review the changes 👁👁

**Comment by phongcao** [comment]
@anderm Looks like there is an issue with the build agent.

**Comment by bengreenier** [comment]
from build agent:

```
LINK : fatal error C1047: The object or library file 'D:\a\1\s\Libraries\UserInterface\..\..\Libraries\WebRTC\Win32\Release\lib\webrtc.lib' was created with an older compiler than other objects; rebuild old objects and libraries [D:\a\1\s\Samples\Client\DirectxWin32\StreamingDirectxClient.vcxproj]
LINK : fatal error LNK1257: code generation failed [D:\a\1\s\Samples\Client\DirectxWin32\StreamingDirectxClient.vcxproj]
```

**Comment by anderm** [comment]
@bengreenier @phongcao I do not see this error on a Win10 machine, I tried on different machines and our build runs successfully. I will re-compile the native library and re-upload on our storage account, I am hoping something went wrong there. Please try to build this on your machine and see if you see any issues.

File: Libraries/ConfigParser/src/config_parser.cpp [hunk 1]
```diff
@@ -3,9 +3,10 @@
 #include <fstream>
 
 #define WIN32_LEAN_AND_MEAN // exclude rarely used windows content
+// #define WEBRTC_EXTERNAL_JSON
```
[the remark this answers is not in the document]
Comment (anderm) [reply Libraries/ConfigParser/src/config_parser.cpp]: Yes, great catch

File: Libraries/SignalingClient/src/ssl_capable_socket.cpp [hunk 2]
```diff
@@ -160,10 +160,10 @@ Socket::ConnState SslCapableSocket::GetState() const
...
 }
 
-int SslCapableSocket::EstimateMTU(uint16_t* mtu)
-{
-	return ssl_adapter_.get() == nullptr ? socket_->EstimateMTU(mtu) : ssl_adapter_->EstimateMTU(mtu);
-}
+//int SslCapableSocket::EstimateMTU(uint16_t* mtu)
+//{
+//	return ssl_adapter_.get() == nullptr ? socket_->EstimateMTU(mtu) : ssl_adapter_->EstimateMTU(mtu);
+//}
```
[the remark this answers is not in the document]
Comment (anderm) [reply Libraries/SignalingClient/src/ssl_capable_socket.cpp]: This is not needed with M62, I will remove it

File: Libraries/WebRTC/3dtoolkit_upgrades.patch [hunk 3]
```diff
@@ -1650,10 +1059,93 @@ index 84bfafb8b..a894822ef 100644
...
+-    codec_specific.codecType = kVideoCodecH264;
+-    codec_specific.codecSpecific.H264.packetization_mode = packetization_mode_;
+-    encoded_image_callback_->OnEncodedImage(encoded_image_, &codec_specific,
+-                                            &frag_header);
+-  }
+-  return WEBRTC_VIDEO_CODEC_OK;
 +NVENCSTATUS H264EncoderImpl::CheckDeviceNVENCCapability()
 +{
++	auto nvpipe = LoadLibrary(L"Nvpipe.dll");
++	if (nvpipe == NULL) {
```
[the remark this answers is not in the document]
Comment (anderm) [reply Libraries/WebRTC/3dtoolkit_upgrades.patch]: That is correct, I will update the Wiki once this is merged.

**Comment by anderm** [comment]
Build agent is back online after library re-upload. Merging now

**Pull request merged**